dynamic_env_validator.py

```diff
@@ -0,0 +1,48 @@
+import os
+
+import yaml
+from pydantic import EmailStr, Field, create_model
+from pydantic_settings import BaseSettings
+
+
+class FileStr(str):
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, value, field):
+        if not os.path.exists(value):
+            raise ValueError(f"File {value} doesn't exist.")
+        return value
+
+
+TYPE_MAPPING = {
+    "str": str,
+    "int": int,
+    "float": float,
+    "bool": bool,
+    "EmailStr": EmailStr,
+    "FileStr": FileStr,  # Add FileStr here
+}
+
+
+def load_yaml(filename: str) -> dict:
+    with open(filename, "r") as f:
+        return yaml.safe_load(f)
+
+
+def create_dynamic_settings(yaml_filename: str) -> BaseSettings:
+    config = load_yaml(yaml_filename)
+    fields = {}
+    for name, value in config.items():
+        field_info = value.get("field_info", {})
+        field_type = TYPE_MAPPING.get(value.get("type"))
+        fields[name] = (field_type, Field(**field_info))
+
+    return create_model("DynamicSettings", __base__=BaseSettings, **fields)
+
+
+DynamicSettings = create_dynamic_settings("env.yaml")
+settings = DynamicSettings(_env_file=".env")
```

Comment:
I think we misunderstood each other. the .env should remain :

ENVIRONMENT_VARIABLE_NAME=ENVIRONMENT_VARIABLE_VALUE

Only the validation specification (env.yaml) should be in yaml.

Reply:
Not sure I understand. This needs both .env and env.yaml to work.

Reply:
How does this work when they are no .env and we are setting instead environment variables through docker run (or equivalent container runner)?